File: advertisingonrails/app/models/deal.rb

class Deal < ApplicationRecord
  enum DealType: [:OpenAuction, :PrivateAuction, :PreferredDeal, :ProgrammaticGuaranteed]


  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  has_many :Publisher, class_name: 'Publisher'
  has_many :InventorySources, class_name: 'InventorySource'
  has_many :Placements, class_name: 'Placement'

end

File: advertisingonaspdotnet/Domain/Deal.cs

using advertisingonaspdotnet.Contracts;

namespace advertisingonaspdotnet.Domain;

public class Deal
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public virtual long? DealId { get; set; }
    public virtual Money? FloorPrice { get; set; }
    public virtual Publisher? Publisher { get; set; }
    public virtual ICollection<InventorySource> InventorySources { get; set; } = new List<InventorySource>();
    public virtual ICollection<Placement> Placements { get; set; } = new List<Placement>();
    public virtual DealType? DealType { get; set; }

    public static Deal FromRequest(DealRequest request)
    {
        return new Deal
        {
            Id = request.Id,
            FloorPrice = request.FloorPrice,
            DealType = request.DealType,
        };
    }
}
